smarts/env/gymnasium/wrappers/metrics.py

```diff
@@ -109,7 +109,8 @@ def step(self, action: Dict[str, Any]):
                 dones["__all__"] = True
             dones.update({a: d["done"] for a, d in info.items()})
 
-        obs = {agent_id: o for agent_id, o in obs.items() if o["active"]}
+        if len(obs) > 0 and isinstance(next(iter(obs.values())), dict):
```

Comment:
`len(obs)` is checked to be greater than zero on line 101 above.

Reply:
Right thanks for pointing that out.